src/hott/07-fibers.rzk.md

```diff
@@ -271,3 +271,157 @@ work from the assumption that `f` is a half adjoint equivalence.
+  (a : A)
+  (b : B a)
+  : homotopy-fiber-strict-fiber a
+    ( strict-fiber-homotopy-fiber a ((a, b), refl)) =
+    ( (a, b), refl)
+  := refl
+
+#def retract-strict-fiber-homotopy-fiber
+  (a : A)
+  (((a', b'), p) : fib (total-type A B) A (total-type-projection A B) a)
+  : homotopy-fiber-strict-fiber a (strict-fiber-homotopy-fiber a ((a', b'), p))
+    = ((a', b'), p)
+  :=
+    ind-fib
+    ( total-type A B)
+    ( A)
+    ( total-type-projection A B)
+    ( \ a0 ((a'', b''), p') →
+      homotopy-fiber-strict-fiber a0
+      ( strict-fiber-homotopy-fiber a0 ((a'', b''), p')) = ((a'', b''), p'))
+    ( \ (a'', b'') → refl)
+    ( a)
+    ( ((a', b'), p))
+
+#def equiv-homotopy-fiber-strict-fiber
+  (a : A)
+  : Equiv
+    ( B a)
+    ( fib (total-type A B) A (total-type-projection A B) a)
+  :=
+    ( homotopy-fiber-strict-fiber a,
+      ( ( strict-fiber-homotopy-fiber a,
+          retract-homotopy-fiber-strict-fiber a),
+        ( strict-fiber-homotopy-fiber a,
+          retract-strict-fiber-homotopy-fiber a)))
+
+#end strict-vs-homotopy-fiber
+```
+
+## Fibers of composites
+
+The fiber of a composite function is a sum over the fiber of the second function
+of the fibers of the first function.
+
+```rzk
+#section fiber-composition
+
+#variables A B C : U
+#variable f : A → B
+#variable g : B → C
+
+#def fiber-sum-fiber-comp
+  (c : C)
+  ((a, r) : fib A C (comp A B C g f) c)
+  : ( Σ ((b, q) : fib B C g c), fib A B f b)
+  := ((f a, r), (a, refl))
+
+#def fiber-comp-fiber-sum
+  (c : C)
+  ( ((b, q), (a, p)) : Σ ((b, q) : fib B C g c), fib A B f b)
+  : fib A C (comp A B C g f) c
+  := (a, concat C (g (f a)) (g b) c (ap B C (f a) b g p) q)
+
+#def retract-fiber-sum-fiber-comp
+  (c : C)
+  ((a, r) : fib A C (comp A B C g f) c)
+  : fiber-comp-fiber-sum c (fiber-sum-fiber-comp c (a, r)) = (a, r)
+  :=
+    eq-eq-fiber-Σ
+    ( A)
+    ( \ a0 → (g (f a0)) = c)
+    ( a)
+    ( concat C (g (f a)) (g (f a)) c refl r)
+    ( r)
+    ( left-unit-concat C (g (f a)) c r)
+
+#def retract-fiber-comp-fiber-sum'
+  (c : C)
+  ((b, q) : fib B C g c)
+  : ((a, p) : fib A B f b) →
+    fiber-sum-fiber-comp c (fiber-comp-fiber-sum c ((b, q), (a, p))) =
+    ((b, q), (a, p))
+  :=
+    ind-fib B C g
+    ( \ c' (b', q') → ((a, p) : fib A B f b') →
+      fiber-sum-fiber-comp c' (fiber-comp-fiber-sum c' ((b', q'), (a, p))) =
+      ((b', q'), (a, p)))
+    ( \ b0 (a, p) →
+      ( ind-fib A B f
+        ( \b0' (a', p') →
+          fiber-sum-fiber-comp (g b0')
+          ( fiber-comp-fiber-sum (g b0') ((b0', refl), (a', p'))) =
+          ((b0', refl), (a', p')))
+        ( \a0 → refl)
+        ( b0)
+        ( (a, p))))
+    ( c)
+    ( (b, q))
+
+#def retract-fiber-comp-fiber-sum
+  (c : C)
+  ( ((b, q), (a, p)) : Σ ((b, q) : fib B C g c), fib A B f b)
+  : fiber-sum-fiber-comp c (fiber-comp-fiber-sum c ((b, q), (a, p))) =
+    ((b, q), (a, p))
+  := retract-fiber-comp-fiber-sum' c (b, q) (a, p)
+
+#def equiv-fiber-comp-fiber-sum
```

Comment:
Over at agda-unimath, we write this in the opposite order, i.e. `equiv-fiber-sum-fiber-comp`, since the underlying map is `fiber-sum-fiber-comp` and not `fiber-comp-fiber-sum`. This appropriately follows the principle of "conclusion first".